DEPRECATION.md

```diff
@@ -19,55 +26,76 @@ underscore (`_`).
 
 The guiding principles are:
 
-- we must not remove or change code without active warnings for least three
-  months or two complete version cycles;
+- removals or behavior changes in the public API can only occur in major releases;
+
+- new deprecations to the public API can only occur in minor releases;
 
 - there must always be a way to achieve valid goals that does not issue any
-  warnings;
+  warnings with the most recent two minor releases in a series;
+
+- never assume that an object that is part of the public interface is not in use.
+
+While the no-breaking-changes rule is only formally required *within* a major release series, you should make every effort to avoid breaking changes wherever possible.
+Similarly, while it is permissible where necessary for behavior to change with no single-code path to support both the last minor of one major release and the first minor of a new major release, it is still strongly preferable if you can achieve this.
```

Comment:
I think this should be *single code path*. For example *two-code path* doesn't make sense but *two code paths* does.

Reply:
I think I using the hyphen to make it so that in your example, the plural would have been "two-codes path", as in "the direction which requires two codes" - "path" wasn't meant in the sense of "codepath" but "direction". Happier for better wording, though.

Reply:
That *does* make sense.  *single code path* in the sense of "sequence of instructions" does not in this context.

Reply:
I'm struggling a bit to get the precise meaning of this
> behavior [to] change with no single-code path to support both the last minor of one major release and the first minor of a new major release

* So *single-code path* means a solution such that the user has one chunk of code that they don't need to alter between the two versions?
* Does *behavior change* here refer to a change in behavior without a change in API? In other words, a user runs code in version A. Then in version B, the same code runs but gives different results.
* Suppose the answer to these questions is "yes" (and say the dev insists on maintaining both) Then I don't see a way to turn this into a non-breaking change.

Reply:
Yeah, in my reading this is describing the situation of 0.46 ->1.0 where we introduced a deprecation warning in 0.46 and removed that api in 1.0. This is a breaking change and because it's for a new major version this is allowed. This is basically just saying, that while the policy allows us to to do with the major version bump, it's preferable to try and maintain compatibility if possible.

Reply:
Yeah, that's what I was going for: "you're technically _allowed_ to make changes that don't have a single way that works on both 0.x and 1.0, but really try not to do that". If there's better wording people want, I'm happy to update.